core/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -16,19 +16,45 @@
 
 package org.apache.spark.sql.delta
 
-import org.apache.spark.sql.delta.actions.Metadata
+import scala.collection.mutable.ArrayBuffer
+import scala.util.control.NonFatal
+
+import org.apache.spark.sql.delta.DeltaParquetFileFormat._
+import org.apache.spark.sql.delta.actions.{DeletionVectorDescriptor, Metadata}
+import org.apache.spark.sql.delta.deletionvectors.DeletedRowsMarkingFilter
 import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
 
+import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.execution.datasources.PartitionedFile
 import org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat
+import org.apache.spark.sql.execution.vectorized.{OffHeapColumnVector, OnHeapColumnVector, WritableColumnVector}
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.sources.Filter
-import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.types.{ByteType, StructField, StructType}
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+import org.apache.spark.util.SerializableConfiguration
 
-/** A thin wrapper over the Parquet file format to support columns names without restrictions. */
-class DeltaParquetFileFormat(metadata: Metadata) extends ParquetFileFormat {
+/**
+ * A thin wrapper over the Parquet file format to support
+ *  - columns names without restrictions.
+ *  - populated a column from the deletion vector of this file (if exists) to indicate
+ *    whether the row is deleted or not according to the deletion vector. Consumers
+ *    of this scan can use the column values to filter out the deleted rows.
+ */
+class DeltaParquetFileFormat(
+    metadata: Metadata,
+    val isSplittable: Boolean = true,
+    val disablePushDowns: Boolean = false,
+    val tablePath: Option[String] = None,
+    val broadcastDvMap: Option[Broadcast[Map[String, DeletionVectorDescriptor]]] = None,
+    val broadcastHadoopConf: Option[Broadcast[SerializableConfiguration]] = None)
+  extends ParquetFileFormat {
+  // Validate either we have all arguments for DV enabled read or none of them.
+  require(!(broadcastHadoopConf.isDefined ^ broadcastDvMap.isDefined ^ tablePath .isDefined ^
```

Comment:
nit: `tablePath .isDefined` there's a space there? is that valid syntax?

Reply:
Will remove the extra space. It is valid from the compiler perspective.